### PR TITLE
[TypeDeclaration][Php 8] Enable ReturnTypeDeclarationRector

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: packages/BetterPhpDocParser/PhpDocParser/StaticDoctrineAnnotationParser.php [hunk 1]
```diff
@@ -101,9 +108,11 @@ private function resolveAnnotationValues(BetterTokenIterator $tokenIterator): ar
     }
 
     /**
-     * @return bool|int|mixed|mixed[]|string
+     * @return array<mixed>
      */
-    private function parseValue(BetterTokenIterator $tokenIterator)
+    private function parseValue(
+        BetterTokenIterator $tokenIterator
+    ): CurlyListNode | string | array | ConstExprFalseNode | ConstExprTrueNode | ConstExprIntegerNode | DoctrineAnnotationTagValueNode
```
Comment (TomasVotruba) [inline packages/BetterPhpDocParser/PhpDocParser/StaticDoctrineAnnotationParser.php]: The `@return array<mixed>` suggest an unclear array.
But return type shows much more. I think the `@return` annotation should be removed, as it does not add any value here. The `mixed[]` and `array` have the same information value

**Comment by TomasVotruba** [comment]
Thank you, this looks solid :+1: 

I've added a comments, but the rest looks ready

File: packages/BetterPhpDocParser/PhpDocParser/StaticDoctrineAnnotationParser.php (hunk 1 above)
Comment (samsonasik) [reply packages/BetterPhpDocParser/PhpDocParser/StaticDoctrineAnnotationParser.php]: Without `array` doc with type mixed specified, it make PHPStan error notice in previously https://github.com/rectorphp/rector-src/runs/2774592028#step:5:44

Do you have idea?

Comment (TomasVotruba) [reply packages/BetterPhpDocParser/PhpDocParser/StaticDoctrineAnnotationParser.php]: I wish there was some extension that ignores this message if type is union type.

Comment (samsonasik) [reply packages/BetterPhpDocParser/PhpDocParser/StaticDoctrineAnnotationParser.php]: It seems we can do with: 

```neon
# phpstan.neon
parameters:
	checkMissingIterableValueType: false
```

https://phpstan.org/blog/solving-phpstan-no-value-type-specified-in-iterable-type

what do you think?

Comment (samsonasik) [reply packages/BetterPhpDocParser/PhpDocParser/StaticDoctrineAnnotationParser.php]: @TomasVotruba I've set `checkMissingIterableValueType: false` to `phpstan.neon` https://github.com/rectorphp/rector-src/pull/184/commits/db92d4d9e12525ffbbe20182ac0270c12f81a761

Comment (TomasVotruba) [reply packages/BetterPhpDocParser/PhpDocParser/StaticDoctrineAnnotationParser.php]: That would disable it for every array, even

```php
function getItems(): array
{
   // ...
}
```

There we want to have specific type in `@return string[]`

Comment (samsonasik) [reply packages/BetterPhpDocParser/PhpDocParser/StaticDoctrineAnnotationParser.php]: Ok, reverted back. Do you have idea to handle that?

Comment (staabm) [reply packages/BetterPhpDocParser/PhpDocParser/StaticDoctrineAnnotationParser.php]: Maybe open a upstream phpstan issue?

**Comment by TomasVotruba** [comment]
This looks good... ready to merge?

**Pull request merged**